leasing/report/lease/extra_city_rent.py

```diff
@@ -25,6 +22,72 @@ def get_recipient_address(obj):
     ]))
 
 
+def get_contract_number(obj):
+    contract_numbers = []
+    for contract in obj.contracts.all():
+        if not contract.contract_number:
+            continue
+
+        contract_numbers.append(contract.contract_number)
+
+    return ' / '.join(contract_numbers)
+
+
+def get_lease_area_identifier(obj):
+    lease_area_identifiers = []
+
+    for lease_area in obj.lease_areas.all():
+        if lease_area.archived_at:
+            continue
+
+        lease_area_identifiers.extend([lease_area.identifier])
+
+    return ' / '.join(lease_area_identifiers)
+
+
+def get_address(obj):
+    addresses = []
+
+    for lease_area in obj.lease_areas.all():
+        if lease_area.archived_at:
+            continue
+
+        for area_address in lease_area.addresses.all():
+            if not area_address.is_primary:
+                continue
+
+            addresses.append(area_address.address)
+
+    return ' / '.join(addresses)
+
+
+def get_tenants(obj):
+    today = datetime.date.today()
+
+    contacts = set()
+
+    for tenant in obj.tenants.all():
+        for tc in tenant.tenantcontact_set.all():
+            if tc.type != TenantContactType.TENANT:
+                continue
+
+            if (tc.end_date is None or tc.end_date >= today) and (tc.start_date is None or tc.start_date <= today):
+                contacts.add(tc.contact)
+
+    return ', '.join([c.get_name() for c in contacts])
+
+
+def get_total_area(obj):
+    total_area = 0
+    for lease_area in obj.lease_areas.all():
+        if lease_area.archived_at:
```

Comment:
Have to check this with customer. I'm pretty sure that if `lease_area.archived_at` is not `None`, that means that it's a historical area but one that's stated in the lease contract, which means that it's actually more "real" than the areas that exist now.

Reply:
Pretty sure was wrong. It's actually `lease_area.in_contract` that preserves the area as a historical, contractual one.